shared/dataflow/codeql/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -891,6 +891,8 @@ module MakeImplCommon<LocationSig Location, InputSig<Location> Lang> {
       nodeDataFlowType(this.asNode(), result)
       or
       nodeDataFlowType(this.asParamReturnNode(), result)
+      or
+      isTopType(result) and this.isImplicitReadNode(_)
```

Comment:
Are we certain that `isTopType` always has results?

Reply:
Reasonably sure, but not completely. There are two options: Either we make it part of the language-specific input, or we make it a consistency check. I think I like the latter to avoid add yet another predicate in the input.

Reply:
A consistency check sounds fine to me.

Reply:
Found to fail on C# with
```
DataFlowType should have one top type that's compatible with all other types, but has 3.
```
That's harmless, but not intentional, so agreed to add this to the input signature instead as a followup.